### ISSUE TITLE
Magento v2.1.2: Check Component Dependency We found conflicting component dependencies.

### ISSUE THREAD

**Issue by bbtimx** [magento/magento2#7917]
### Preconditions

1.  Magento v 2.1.2
2.  PHP 5.6

### Steps to reproduce

1. Install an module from Magento Store
2. Component Manager
3. Install Store Module
4. Readiness Check

### Expected result

1. Continue to Step 2 to install the module
2. The  Readiness Check stuck on error: 
```
Check Component Dependency
We found conflicting component dependencies. Hide detail

For additional assistance, see component dependency help .
```

### Actual result

1. [Screenshot, logs]
![image](https://cloud.githubusercontent.com/assets/4718129/21401863/31b83af2-c7b6-11e6-95df-83ef7b840c9d.png)

![image](https://cloud.githubusercontent.com/assets/4718129/21401997/cc8519ba-c7b6-11e6-95c6-7111d6fdab1f.png)

The: **http://xxxxxx.de/setup/index.php/dependency-check/component-dependency 500 (Internal Server Error)** tell me:

```
An error occurred during execution; please try again later.

Additional information:

Exception

File:
/var/www/share/xxxxxx.de/htdocs/vendor/magento/framework/App/ErrorHandler.php:61
Message:
Warning: Invalid argument supplied for foreach() in /var/www/share/xxxxxx.de/htdocs/setup/src/Magento/Setup/Controller/DependencyCheck.php on line 73
Stack trace:
#0 /var/www/share/xxxxxx.de/htdocs/setup/src/Magento/Setup/Controller/DependencyCheck.php(73): Magento\Framework\App\ErrorHandler->handler(2, 'Invalid argumen...', '/var/www/share/...', 73, Array)
#1 /var/www/share/xxxxxx.de/htdocs/vendor/zendframework/zend-mvc/src/Controller/AbstractActionController.php(82): Magento\Setup\Controller\DependencyCheck->componentDependencyAction()
#2 [internal function]: Zend\Mvc\Controller\AbstractActionController->onDispatch(Object(Zend\Mvc\MvcEvent))
#3 /var/www/share/xxxxxx.de/htdocs/vendor/zendframework/zend-eventmanager/src/EventManager.php(444): call_user_func(Array, Object(Zend\Mvc\MvcEvent))
#4 /var/www/share/xxxxxx.de/htdocs/vendor/zendframework/zend-eventmanager/src/EventManager.php(205): Zend\EventManager\EventManager->triggerListeners('dispatch', Object(Zend\Mvc\MvcEvent), Object(Closure))
#5 /var/www/share/xxxxxx.de/htdocs/vendor/zendframework/zend-mvc/src/Controller/AbstractController.php(118): Zend\EventManager\EventManager->trigger('dispatch', Object(Zend\Mvc\MvcEvent), Object(Closure))
#6 /var/www/share/xxxxxx.de/htdocs/vendor/zendframework/zend-mvc/src/DispatchListener.php(93): Zend\Mvc\Controller\AbstractController->dispatch(Object(Zend\Http\PhpEnvironment\Request), Object(Zend\Http\PhpEnvironment\Response))
#7 [internal function]: Zend\Mvc\DispatchListener->onDispatch(Object(Zend\Mvc\MvcEvent))
#8 /var/www/share/xxxxxx.de/htdocs/vendor/zendframework/zend-eventmanager/src/EventManager.php(444): call_user_func(Array, Object(Zend\Mvc\MvcEvent))
#9 /var/www/share/xxxxxx.de/htdocs/vendor/zendframework/zend-eventmanager/src/EventManager.php(205): Zend\EventManager\EventManager->triggerListeners('dispatch', Object(Zend\Mvc\MvcEvent), Object(Closure))
#10 /var/www/share/xxxxxx.de/htdocs/vendor/zendframework/zend-mvc/src/Application.php(314): Zend\EventManager\EventManager->trigger('dispatch', Object(Zend\Mvc\MvcEvent), Object(Closure))
#11 /var/www/share/biokamine.de/htdocs/setup/index.php(31): Zend\Mvc\Application->run()
#12 {main}
```

**Comment by philwinkle** [comment]
I'm having this issue too. Any word?

**Comment by sjiuk** [comment]
philwinkle did you resolve this problem?

**Comment by moejoee91** [comment]
any news?
did u solve it ? i'm having the same exact issue with 2.1.6

**Comment by rogmoreno** [comment]
i solved this with max_execution_time = 0  on php.ini

**Comment by kskarim** [comment]
I have the same problem thank you for your help urgent  ????

**Comment by lano-vargas** [comment]
I've installed a fresh installation 2.1.7 tried to install MailChimp extension I've got same problem here.
Need input Please. Thank's for the urgency appreciate.

**Comment by orlangur** [comment]
@lano-vargas, GitHub is intended for bug reports, please use [Magento Stack Exchange](https://magento.stackexchange.com) for questions.

Try installing this extension using CLI, in such way you'll what are the actual dependency conflicts.

**Comment by lano-vargas** [comment]
@orlangur Thank's.
I am sure your intentions are good... but I didn't ask a question there are no question marks.
And yes I do have trouble installing any other extensions.
using CLI is how I normally install other extensions however I am following MailChimp recommendations ref: https://kb.mailchimp.com/integrations/e-commerce/connect-or-disconnect-mailchimp-for-magento-2.

**Comment by orlangur** [comment]
@lano-vargas ok, let me rephrase - any problem which is not reproducible on clean Magento installation is out-of-scope here.

If you experience problem with installing some third-party extension, its vendor should be contacted first and only after a bug is confirmed on Magento side it should be reported here (I really doubt there could be such bugs in extension installation mechanism as it is just a wrapper for Composer).

**Comment by lano-vargas** [comment]
@orlangur, I've just created a new installation of Magento just to make sure that my existing Magento isn't causing the issue and buggy, but let me try again please. So yes I did try to install other modules on other existing installation following exactly as bbtimx referencing and the same issue occurs.
After all the question then is this a bug ?  Should I expect this to happen on a new installation? Thank's

**Comment by orlangur** [comment]
Based on https://github.com/magento/magento2/issues/7917#issuecomment-312554412 I assume the problem could occur in particular server configuration. Generally seeing `500 Internal Server Error` one should identify a problem precisely on his side.

**Comment by lano-vargas** [comment]
@orlangur  I am afraid I've tried it didn't do!

**Comment by orlangur** [comment]
This issue does not contain enough information to reproduce it by anyone else. Anyone facing it, please add more details performing some investigation on your side and I will be glad to reopen it.

**Comment by fsrojash** [comment]
Hi there, i have the same error on a fresh installation of Magento 2.1.13 on PHP 5.6. Any thoughts around it?

**Comment by shorav13** [comment]
Any solution guys??Facing same issue

**Comment by Jeeva-Rathinam** [comment]
Any one found solution?

**Comment by orlangur** [comment]
@fsrojash @shorav13 @Jeeva-Rathinam in order to investigate this issue clear steps to reproduce needs to be provided first.

**Comment by dandelionred** [comment]
@orlangur Same problem here in 2.3.0.

The reason for error 500 on /setup/index.php/dependency-check/component-dependency is memory limit hit. I use the stock magento nginx snippet hence memory limit was 756M.

I tried increasing memory limit to 1756M and here is what see:
1. The php-fpm process handling the request sits at low ram and low cpu usage for some times (it downloads something according to strace).
2. At some point cpu usage jumps to 100% and ram usage slowy goes up to around 1G (1.089 .. 1.092) and stays there.
3. Next the process sits at 100% cpu usage, strace output is empty.
4. After like 4 or 5 mins (4x3.1ghz i5 cpu, ddr3 ram) the request finishes successfully.

So there are 2 problems:
- huge ram usage exceeding the limit set in the stock magento nginx snippet
- ridiculously cpu-heavy dependency checking code

**Comment by orlangur** [comment]
@dandelionred memory limit must be 2Gb according to https://devdocs.magento.com/guides/v2.3/install-gde/system-requirements-tech.html.

> ridiculously cpu-heavy dependency checking code

It's Composer, dude, and such behavior is correct :)

**Comment by dandelionred** [comment]
@orlangur There is still a _small_ problem. Default memory limit is 756 https://github.com/magento/magento2/blob/d7ac52bbbc11e35aac5f87f0ff0fd6232d75e79b/nginx.conf.sample#L43. The only report a user gets when trying to install an extension is "Check Component Dependency = We found conflicting component dependencies." without any extra details.

![screenshot from 2019-01-11 23 10 31](https://user-images.githubusercontent.com/19490344/51059874-5c86ff80-15f6-11e9-93af-9cb4c3c54a10.png)

It doesn't tell the reason why it reported about a conflict was a http 500 response to /setup/index.php/dependency-check/component-dependency request. Don't you think it is kinda not user friendly/confusing?

**Comment by orlangur** [comment]
@dandelionred default memory limit is correct according to https://devdocs.magento.com/guides/v2.3/install-gde/prereq/php-settings.html.

However, setup wizard in this scenario obviously belongs to 

> Installing and updating Magento components from Magento Marketplace, 2G

So, I believe there must be at least some warning in case of lower memory limit.

**Comment by dandelionred** [comment]
@orlangur 
> default memory limit is correct according to 

The string from nginx.conf.sample I referenced above was inside `location ~* ^/setup($|/) {}`. Hence the stock nginx config sample suggests 756M for `/setup/` even though the doc says 2G for _Installing and updating Magento components_. Since dependency check takes over 1G the thing doesn't work at all out-of-the-box.

> So, I believe there must be at least some warning in case of lower memory limit.

I'm sure the main problem here is there is no feedback on the server-side error. It treats anything but OK as "We found conflicting component dependencies".

Back to the heavy cpu usage during dependency check
> It's Composer, dude, and such behavior is correct :)

Strangely enough, installing an extension with `composer require ...` has no such problems. You said [above](https://github.com/magento/magento2/issues/7917#issuecomment-329778426) the web UI for managing extensions is a wrapper around composer. The wrapper is probably wrapped too much!

**Comment by orlangur** [comment]
@dandelionred,

> nginx config sample suggests 756M for /setup/ even though the doc says 2G for Installing and updating Magento components

Being inside Setup Wizard does not necessary mean you perform heavy Composer operations like installing components.

> Strangely enough, installing an extension with `composer require` ... has no such problems

I'm pretty sure this is not true, but to say for sure we need to look into profiler report.

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-nazar. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [ ] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [ ] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [ ] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [ ] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 6. Add label `Issue: Confirmed` once verification is complete. 

- [ ] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by ghost** [comment]
HI @bbtimx thank you for you report, I closing this issue as we have this one https://github.com/magento/magento2/issues/20684

**Comment by orlangur** [comment]
@engcom-backlog-nazar mentioned issue DOES NOT contain any details. This issue should not be closed until there is an evidence they have the same root cause.

**Comment by ghost** [comment]
@orlangur Yes i agree, but this looks like we have 2 duplicate issue.

**Comment by orlangur** [comment]
@engcom-backlog-nazar we will always be able to close redundant one, for now only symptom is the same. Current issue contains useful details.

**Comment by mupamupa** [comment]
Any progress with this issue?

**Comment by dandelionred** [comment]
@mupamupa

Have you tried raising the memory limit to 2G?

**Comment by ghost** [comment]
HI @bbtimx @orlangur i'm not able to reproduce this error on 2.3.1 only have issue from ->https://github.com/magento/magento2/issues/20684
![DeepinScreenshot_select-area_20190329155201](https://user-images.githubusercontent.com/40756439/55238231-a1592480-523c-11e9-9e5c-720de5267203.png)

**Comment by ghost** [comment]
Hi folks, i'm closing this as this now not reproducing at 2.3-develop branch, have only issue with -> https://github.com/magento/magento2/issues/20684 , feel free to reopen if issue still exist.

**Comment by dandelionred** [comment]
@engcom-backlog-nazar I explained the whole problem behind the issue in https://github.com/magento/magento2/issues/7917#issuecomment-453583809, https://github.com/magento/magento2/issues/7917#issuecomment-453658759 and https://github.com/magento/magento2/issues/7917#issuecomment-453690781

Nothing has changed on the 2.3-develop branch:
- .htaccess still has
  `php_value memory_limit 756M`
- nginx.conf.sample still has
  `fastcgi_param  PHP_VALUE "memory_limit=756M \n max_execution_time=600";`

**Comment by nittn** [comment]
Solved the same issue
local php values in .htaccess and .user.ini were overriding master php values in php.ini

increased php local memory_limit in .htaccess and .user.ini to 2G
php master memory_limit was already 2G

hope this helps someone.

**Comment by TechnoBuddhist** [comment]
@nittn THANK YOU!! This resolved my issue! I successfully upgraded from 2.3.0 -> 2.3.1.

I noticed that .htaccess and .user.ini are still set to 756M in 2.3.1. Unless there is a good reason for them to be this then I would suggest devs to make the change. I won't make a pull request myself as I don't want to encroach on something I've got no knowledge of!!

**Comment by gfrey-namcopool** [comment]
I can confirm increasing memory to 2G still solves this issue (2.3.1 -> 2.3.4)
